packages/react-components/react-rating-preview/src/components/RatingItem/RatingItem.test.tsx

```diff
@@ -1,10 +1,108 @@
-  it('renders a default state', () => {
-    const result = render(<RatingItem>Default RatingItem</RatingItem>);
-    expect(result.container).toMatchSnapshot();
+  isConformant({
+    Component: RatingItem,
+    displayName: 'RatingItem',
+    // Need to disable this test because certain slots are not rendered without specific context values
+    disabledTests: ['component-has-static-classnames-object'],
+  });
+  it('does not render input elements when interactive is false', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: false,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 1,
+      size: 'medium',
+    };
+    const { getByTestId } = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={2} data-testid="test" />
+      </RatingItemProvider>,
+    );
+    const items = getByTestId('test');
+    expect(items.getElementsByTagName('input').length).toEqual(0);
+  });
+  it('renders only the full value input when step is 1 and interactive is true', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: true,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 1,
+      size: 'medium',
+    };
+    const { getByTestId } = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={2} data-testid="test" />
+      </RatingItemProvider>,
+    );
+    const items = getByTestId('test');
+    expect(items.getElementsByTagName('input').length).toEqual(1);
+  });
+  it('renders both the full value input and the half value input when step is 0.5 and interactive is true', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: true,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 0.5,
+      size: 'medium',
+    };
+    const { getByTestId } = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={2} data-testid="test" />
+      </RatingItemProvider>,
+    );
+    const items = getByTestId('test');
+    expect(items.getElementsByTagName('input').length).toEqual(2);
+  });
+  it('sets the half value input to checked when the value is 2.5', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 2.5,
+      interactive: true,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 0.5,
+      size: 'medium',
+    };
+    const { getByTestId } = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={2.5} data-testid="test" />
+      </RatingItemProvider>,
+    );
+    const items = getByTestId('test');
+    const checkedItems = queryAllByRole(items, 'radio').filter(item => (item as HTMLInputElement).checked);
+    expect(checkedItems.length).toEqual(1);
```

Comment:
Don't need `testid` if you're getting by role. Also, RatingItem's value should be 3 here I think? The value prop should only be a whole number. Finally, probably worthwhile adding an `expect` that the checked radio item's value is the same as the context value.

```suggestion
    const { queryAllByRole } = render(
      <RatingItemProvider value={contextValues}>
        <RatingItem value={3} />
      </RatingItemProvider>,
    );
    const inputs = queryAllByRole('radio' as HTMLInputElement[];
    const checkedInputs= inputs.filter(input => input.checked);
    expect(checkedInputs.length).toEqual(1);
    expect(checkedInputs[0].value).toEqual(contextValues.value);
```